src/desktop/settings.rs

```diff
@@ -181,14 +181,27 @@ impl<'a> Settings<'a> {
     }
 
     /// Listen to changes of the system's preferred color scheme
-    pub async fn receive_color_scheme_changed(
+    ///
+    /// The stream will also initially yield the current color scheme preference and filter out
+    /// Duplicates.
+    pub async fn receive_color_scheme_changed_with_init(
```

Comment:
`duplicates` instead also not really a fan of this function tbh. The user can easily call `self.color_scheme().await?;` and listen to changes afterwards.

Reply:
The alternative to initializing `current` would be using an Option type. I liked the Idea to get the initial value in the stream, because I used the signal to call a function that makes sure everything is setup with the current preference and I don't need to store the preference in a variable. In my case I just had to wait for a changed color-scheme and execute my function and that's it. Would be interesting how other usecases look like. Or if you prefer I do it with the Option (because I believe duplicates are not nice).

Reply:
This is the only remaining "problem". I would suggest we drop this part from the patches and open a separate PR for it so we can get the signal stream fixes in, that would also simplify the review/discussion

Reply:
What would 'drop' mean, removing the function?

Reply:
yes and open a separate patch with only that function